src/c_api.cc

```diff
@@ -120,8 +120,45 @@ inline void Allreduce(void *sendrecvbuf,
     default: utils::Error("unknown enum_op");
   }
 }
-
-
+inline void AllgatherRing(void *sendrecvbuf_,
+                          size_t total_size,
+                          size_t beginIndex,
+                          size_t size_node_slice,
+                          size_t size_prev_slice,
+                          int enum_dtype) {
+  using namespace engine::mpi;
```

Comment:
See above comment, can we move the definition into c header?

Reply:
Done